src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputIncorrectJavadocParagraphCheck.java

```diff
@@ -27,51 +39,84 @@ class InputIncorrectJavadocParagraphCheck {
+    /**<p>Some Javadoc.
+     * <p>
+     * <p><p>
+     * <p>/^WARN/   Some Javadoc.<p>
+     */
+    // 2 violations 3 lines above:
+    //                            '\<p\> tag should be placed immediately before the first word'
+    //                            '\<p\> tag should be preceded with an empty line.'
+    // 2 violations 5 lines above:
+    //                            '\<p\> tag should be placed immediately before the first word'
+    //                            '\<p\> tag should be preceded with an empty line.'
      class InnerInputCorrectJavaDocParagraphCheck {
 
+        // 2 violations 4 lines below:
+        //                            '\<p\> tag should be placed immediately before the first word'
+        //                            '\<p\> tag should be preceded with an empty line.'
         /**
-         * Some Javadoc.<p> //warn
+         * Some Javadoc.<p>
          *
          * @since 8.0
          */
         public static final byte NUL = 0;
 
-        /**<p> //warn
+        // 2 violations 4 lines below:
+        //                            '\<p\> tag should be placed immediately before the first word'
+        //                            'Redundant \<p\> tag.'
+        // violation 4 lines below '\<p\> tag should be placed immediately before the first word'
+        /**<p>
          * /^WARN/ Some Javadoc.
          *
-         * <P> //warn
+         * <P>
          * /^WARN/
-         * <p> //warn
-         *  /^WARN/ Some Javadoc.<p> //warn
+         * <p>
+         *  /^WARN/ Some Javadoc.<p>
          * @see <a href="http://www.gwtproject.org/doc/latest/DevGuideOrganizingProjects.html#DevGuideModules">
          *     Documentation about GWT emulated source</a>
          */
+        // 2 violations 5 lines above:
+        //                            '\<p\> tag should be placed immediately before the first word'
+        //                            '\<p\> tag should be preceded with an empty line.'
+        // 2 violations 7 lines above:
+        //                            '\<p\> tag should be placed immediately before the first word'
+        //                            '\<p\> tag should be preceded with an empty line.'
+        // violation 9 lines above 'Javadoc tag '@see' should be preceded with an empty line.'
         boolean emulated() {return false;}
     }
 
     InnerInputCorrectJavaDocParagraphCheck anon = new InnerInputCorrectJavaDocParagraphCheck() {
 
+        // violation 2 lines below 'Redundant \<p\> tag.'
         /**
-         * <p>Some Javadoc. //warn
+         * <p>Some Javadoc.
          *
          * <p>Some Javadoc.
          *
          * @since 8.0
          */
         public static final byte NUL = 0;
 
+        // 2 violations 5 lines below:
+        //                            '\<p\> tag should be placed immediately before the first word'
+        //                            '\<p\> tag should be preceded with an empty line.'
```

Comment:
It looks like a waste of space, especially for long messages. Can we use 2 spaces as an indent for multiple violations?

example:
```
         // 2 violations 5 lines below:
         //   '\<p\> tag should be placed immediately before the first word'
         //   '\<p\> tag should be preceded with an empty line.'
         // violation 4 lines below '\<p\> tag should be placed immediately before the first word'
```

Same for all other cases.

Reply:
Ok, let's do it.
When we initially created this approach it was convenient to do indentation to far right to make it more clear that such comments are very very special to not mix them.
But let's try +2 indentation.

Reply:
done

https://github.com/checkstyle/checkstyle/blob/cf4fb7f7c11e8680e4ae5021ce971583f967b64b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputIncorrectJavadocParagraph.java#L18-L20

Reply:
also renamed Input files and removed trailing `Check` from their name